python/llm/example/transformers_int4.py

```diff
@@ -46,7 +46,8 @@
         print(output_str)
         print(f'Inference time: {end-st} s')
     elif model_path == 'THUDM/chatglm-6b':
-        model = AutoModel.from_pretrained(model_path, trust_remote_code=True, load_in_4bit=True)
+        model = AutoModel.from_pretrained(model_path, trust_remote_code=True, load_in_4bit=True, low_cpu_mem_usage=True)
+        model.to(torch.float32)
```

Comment:
directly using ChatGLM in transformers on cpu also requires `model.to(torch.float32)` or `model.to(torch.bfloat16)`

Reply:
Can we automatically cast non-INT4 layers to FP32 in our conversion? For BF16, can we use auto mixed precision?

Reply:
Yes.
And using BF16 actually perform worse than fp32 in this case.